cirq-core/cirq/protocols/has_stabilizer_effect_protocol.py

```diff
@@ -62,9 +73,39 @@ def _strat_has_stabilizer_effect_from_unitary(val: Any) -> Optional[bool]:
     2x2 unitaries.
     """
     # Do not try this strategy if there is no unitary or if the number of
-    # qubits is not 1 since that would be expensive.
-    qid_shape = protocols.qid_shape(val, default=None)
-    if qid_shape is None or len(qid_shape) != 1 or not protocols.has_unitary(val):
+    # qubits is greater than 3 since that would be expensive.
+    qid_shape = qid_shape_protocol.qid_shape(val, default=None)
+    if qid_shape is None or len(qid_shape) > 3 or not has_unitary_protocol.has_unitary(val):
```

Comment:
Should we insead check that `qid_shape` is either `(2,)` or `(2, 2)` ? For example, what happens if the operation is defined on 2 or less qudits with dimension greater than 2?

Reply:
done